src/vsr/client.zig

```diff
@@ -30,98 +29,30 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
                 results: []const u8,
             ) void;
 
-            pub const Demux = struct {
-                next: ?*Demux = null,
-                user_data: u128,
-                callback: ?Callback, // Null iff operation=register.
-                event_count: u16,
-                event_offset: u16,
-            };
-
             message: *Message.Request,
-            demux_queue: FIFO(Demux) = .{ .name = null },
+            user_data: u128,
+            callback: Callback,
         };
 
-        pub const Batch = struct {
-            request: *Request,
-            demux: *Request.Demux,
-
-            pub fn slice(batch: Batch) []u8 {
-                // Ensure batch.demux is either the one inlined in the Request or is in demux_queue.
-                assert(batch.request.demux_queue.count > 0);
-                if (constants.verify) {
-                    if (batch.request.demux_queue.count == 1) {
-                        assert(batch.request.demux_queue.peek() == batch.demux);
-                    } else {
-                        assert(batch.request.demux_queue.contains(batch.demux));
-                    }
-                }
-
-                const message = batch.request.message;
-                assert(message.header.command == .request);
+        /// Custom Demuxer which passes through to StateMachine.Demuxer.
+        pub fn DemuxerType(comptime operation: StateMachine.Operation) type {
+            return struct {
```

Comment:
It's used outside of `vsr.Client` (in the unit test below and tb_client Context).

Reply:
So why not move this (and the corresponding unit test) move to tb_client Context's file? It seems weird to have it here since `vsr.Client` is now unaware of batching.

Reply:
It's due to `EchoClient`: its Demuxer must return Events back but StateMachine.Demuxer adjusts & returns Results. Creating a stub StateMachine just for EchoClient, with most of the public decls from the original but a custom Demuxer, seemed more weird. So instead clients have a Demuxer wrapper with separate unit_tests each.